Logger/LogFactory.cs

```diff
@@ -2,10 +2,24 @@
 {
     public class LogFactory
     {
-        public BaseLogger CreateLogger(string className)
+        private string? FilePath;
+
+        public BaseLogger? CreateLogger(string className)
         {
-            
-            return null;
+            if (FilePath == null)
```

Comment:
This works, however it is worth knowing that if you use `== null` against a type that has a custom `==` operator defined, could mean that it doesn't behave as you would expect. To ensure it is always a reference comparison against null you can use `is null`
```suggestion
            if (FilePath is null)
```